### PR TITLE
Serialize records to CSV

### LINKED ISSUE #232: add "Serialize records to csv"
please see [csv cookbook](https://docs.rs/csv/*/csv/cookbook/index.html) if using this resource please remember to attribute original author

example should show both free form serialization and struct serialization with serde
preferably in separate snippets but under one example
 
cc https://github.com/brson/rust-cookbook/issues/212

See [CONTRIBUTING.md](https://github.com/brson/rust-cookbook/blob/master/CONTRIBUTING.md) for details.

### PR DESCRIPTION
**Pull request by alisha17**
Fixes #232

### CODE REVIEW

**Comment by brson** [comment]
Looks good.

Instead of doing two examples under one section, can you split it into two sections please? Call the second 
perhaps "Serialize records to CSV using serde".

**Comment by brson** [comment]
This also needs a rebase because of the merge conflict.

File: src/encoding.md [hunk 1]
```diff
@@ -344,6 +346,105 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-serialize-csv]: #ex-serialize-csv
+<a name="ex-serialize-csv"></a>
+## Serialize records to CSV
+
+[![csv-badge]][csv] [![cat-encoding-badge]][cat-encoding]
+
+CSV writer supports automatic serialization from Rust types into CSV
```
Comment (budziq) [inline src/encoding.md]: ease linkify the csv::Writer identifier

File: src/encoding.md [hunk 2]
```diff
@@ -344,6 +346,105 @@ fn run() -> Result<()> {
...
+records. This example shows how to serialize a Rust tuple.
+
+```rust
+# #[macro_use]
+#extern crate error_chain;
```
Comment (budziq) [inline src/encoding.md]: missing space after #

File: src/encoding.md [hunk 3]
```diff
@@ -344,6 +346,105 @@ fn run() -> Result<()> {
...
+## Serialize records to CSV using Serde
+
+[![csv-badge]][csv] [![serde-badge]][serde] [![cat-encoding-badge]][cat-encoding]
+
+The following example shows how to serialize custom structs as CSV records. For this,
+make use of the [serde] crate.
+
+```rust
+# #[macro_use]
+#extern crate error_chain;
```
Comment (budziq) [inline src/encoding.md]: missing space after #

File: src/encoding.md [hunk 4]
```diff
@@ -344,6 +346,105 @@ fn run() -> Result<()> {
...
+#
+# error_chain! {
+#    foreign_links {
+#        IOError(std::io::Error);
+#        CSVError(csv::Error);
+#    }
+# }
+
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "PascalCase")]
```
Comment (budziq) [inline src/encoding.md]: if the rename is important here please describe why and link to docs in the description if not then please remove it.

File: src/encoding.md [hunk 5]
```diff
@@ -344,6 +346,105 @@ fn run() -> Result<()> {
...
+        place: "Zurich",
+        id: 25,
+    })?;
+    wtr.serialize(Record {
+        name: "James",
+        place: "Hong Kong",
+        id: 87,
+    })?;
+
+    wtr.flush()?;
```
Comment (budziq) [inline src/encoding.md]: ease mention why explicit flushing is important here

File: src/encoding.md [hunk 6]
```diff
@@ -344,6 +346,105 @@ fn run() -> Result<()> {
...
+    name: &'a str,
+    place: &'a str,
+    id: u64,
+}
+
+fn run() -> Result<()> {
+    let mut wtr = csv::Writer::from_writer(io::stdout());
+
+    wtr.serialize(Record {
+        name: "Mark",
```
Comment (budziq) [inline src/encoding.md]: how about providing a constructor for `Record` to make the example less verbose

File: src/encoding.md [hunk 7]
```diff
@@ -344,6 +346,105 @@ fn run() -> Result<()> {
...
+#[derive(Debug, Serialize)]
+#[serde(rename_all = "PascalCase")]
+struct Record<'a> {
+    name: &'a str,
+    place: &'a str,
+    id: u64,
+}
+
+fn run() -> Result<()> {
+    let mut wtr = csv::Writer::from_writer(io::stdout());
```
Comment (budziq) [inline src/encoding.md]: i would mention the key identifiers (not described in previous exame) in description linking to the docs

File: src/encoding.md [hunk 8]
```diff
@@ -344,6 +346,105 @@ fn run() -> Result<()> {
...
+```
+
+[ex-csv-serde]: #ex-csv-serde
+<a name="ex-csv-serde"></a>
+## Serialize records to CSV using Serde
+
+[![csv-badge]][csv] [![serde-badge]][serde] [![cat-encoding-badge]][cat-encoding]
+
+The following example shows how to serialize custom structs as CSV records. For this,
+make use of the [serde] crate.
```
Comment (budziq) [inline src/encoding.md]: I would remove the last sentence and just mention serde in the first sentence.

File: src/encoding.md [hunk 9]
```diff
@@ -344,6 +346,105 @@ fn run() -> Result<()> {
...
+#     foreign_links {
+#         CSVError(csv::Error);
+#         IOError(std::io::Error);
+#    }
+# }
+
+fn run() -> Result<()> {
+    let mut wtr = csv::Writer::from_writer(io::stdout());
+
+    wtr.write_record(&["Name", "Place", "ID"])?;
```
Comment (budziq) [inline src/encoding.md]: please mention the key identifiers in the description linking to the docs

**Comment by brson** [comment]
Thanks @alisha17

**Pull request merged**